mesh.go

```diff
@@ -457,13 +458,17 @@ func NewMeshPart(mesh *Mesh, material *Material) *MeshPart {
 }
 
 func (part *MeshPart) Clone() *MeshPart {
-	newMP := NewMeshPart(part.Mesh, part.Material)
-	newMP.Vertices = []*Vertex{}
-	for _, tri := range part.Triangles {
+	newMP := &MeshPart{
+		Mesh:            part.Mesh,
+		Triangles:       make([]*Triangle, len(part.Triangles)),
+		sortedTriangles: make([]*Triangle, len(part.Triangles)),
+		Material:        part.Material,
+	}
+	for i, tri := range part.Triangles {
 		newTri := tri.Clone()
-		newMP.Triangles = append(newMP.Triangles, newTri)
-		newMP.sortedTriangles = append(newMP.sortedTriangles, newTri)
-		newMP.Vertices = append(newMP.Vertices, tri.Vertices...)
+		newMP.Triangles[i] = newTri
+		newMP.sortedTriangles[i] = newTri
+		newMP.Vertices = tri.Vertices
```

Comment:
Shouldn't this still be newMP.Vertices = append(newMP.Vertices, tri.Vertices...)? Here the mesh part (which is a collection of triangles / vertices under a single material) is directly setting its vertex slice to be a reference to a triangle's vertex list (it needs to be all of them).

Reply:
Wops, good catch. I'll commit a fix.

Reply:
Great - thanks a lot for the fix!